lib/eco/treelexer/lexer.py

```diff
@@ -42,3 +68,68 @@ def match(self, pattern, text):
                 return self.match_star(pattern[0] + "*" + pattern[2:], text[1:])
             return False
         return self.match_one(pattern, text) and self.match(pattern[1:], text[1:])
+
+class RegexParser(object):
+
+    def __init__(self):
+        from grammars.grammars import regex
+        p, l = regex.load()
+        self.incparser = p
+        self.inclexer = l
+
+    def load(self, pattern):
+        from incparser.astree import TextNode
+        from grammar_parser.gparser import Terminal
+        self.incparser.reset()
+        bos = self.incparser.previous_version.parent.children[0]
+        bos.insert_after(TextNode(Terminal(pattern)))
+        self.inclexer.relex(bos)
+        self.incparser.reparse()
+        return self.incparser.last_status
+
+    def ast(self, pattern):
+        self.load(pattern)
+        start = self.incparser.previous_version.parent.children[1].alternate
+        return self.parse(start)
+
+    def parse(self, node):
+        from grammar_parser.bootstrap import ListNode
+        if type(node) is ListNode:
+            return self.parse_list(node)
+        if node.name == "OR":
+            return self.parse_or(node)
+        elif node.name == "GROUP":
+            return self.parse_group(node)
+        elif node.name == "STAR":
+            return self.parse_star(node)
+        elif node.name == "PLUS":
+            return self.parse_plus(node)
+        elif node.name == "CHAR":
+            return self.parse_char(node)
+
+    def parse_list(self, node):
+        if len(node.children) > 1:
+            l = []
```

Comment:
Could be `return [self.parse(c) for c in node.children]` I think.

Reply:
True.